ext/datasource/nacos/nacos.go

```diff
@@ -0,0 +1,115 @@
+package nacos
+
+import (
+	"github.com/alibaba/sentinel-golang/ext/datasource"
+	"github.com/alibaba/sentinel-golang/logging"
+	"github.com/alibaba/sentinel-golang/util"
+	"github.com/nacos-group/nacos-sdk-go/clients/config_client"
+	"github.com/nacos-group/nacos-sdk-go/clients/nacos_client"
+	"github.com/nacos-group/nacos-sdk-go/common/constant"
+	"github.com/nacos-group/nacos-sdk-go/common/http_agent"
+	"github.com/nacos-group/nacos-sdk-go/vo"
+	"github.com/pkg/errors"
+)
+
+var (
+	logger = logging.GetDefaultLogger()
+)
+
+type ConfigParam struct {
+	DataId string
+	Group  string
+}
+
+var clientConfig = constant.ClientConfig{
+	TimeoutMs:      10 * 1000,
+	ListenInterval: 30 * 1000,
+	BeatInterval:   5 * 1000,
+	LogDir:         "/nacos/logs",
+	CacheDir:       "/nacos/cache",
+}
+
+type NacosDataSource struct {
+	datasource.Base
+	serverConfig  constant.ServerConfig
+	clientConfig  constant.ClientConfig
+	client        *config_client.ConfigClient
+	isInitialized util.AtomicBool
+	configParam   ConfigParam
+	listener      *vo.ConfigParam
+}
+
+func NewNacosDataSource(clientConfig constant.ClientConfig, serverConfig constant.ServerConfig, configParam ConfigParam, handlers ...datasource.PropertyHandler) (*NacosDataSource, error) {
+	if len(serverConfig.IpAddr) <= 0 || serverConfig.Port <= 0 {
+		return nil, errors.New("The nacos serverConfig is incorrect.")
+	}
+	var ds = &NacosDataSource{
+		clientConfig: clientConfig,
+		serverConfig: serverConfig,
+		configParam:  configParam,
+	}
+	for _, h := range handlers {
+		ds.AddPropertyHandler(h)
+	}
+	return ds, nil
+}
+
+func (s *NacosDataSource) Initialize() error {
+	if !s.isInitialized.CompareAndSet(false, true) {
+		return nil
+	}
+	nc, err := buildNacosClient(s)
+	client, err := config_client.NewConfigClient(nc)
```

Comment:
From my perspective, I think it is up to the user to build the client.
Sentinel shouldn't instantiate the client, instead, check the client's legitimacy.